src/main/kotlin/cli/Scrape.kt

```diff
@@ -7,25 +7,89 @@ import com.github.ajalt.clikt.parameters.types.int
 import models.SubjectCode
 import models.Term
 import mu.KotlinLogging
-import services.JsonMapper
-import services.scrapeFromCatalog
-import services.scrapeAllFromCatalog
+import scraping.models.Course
+import services.*
 import utils.writeToFileOrStdout
 
 // TODO Change this to package-level protected if that becomes a thing
 /**
- * CLI for querying NYU Albert. Has two subcommands, `catalog` and `section`.
+ * CLI for querying NYU Albert. Has three subcommands, `catalog` and `section` and 'sections'.
  * @author Albert Liu
  */
 internal class Scrape : CliktCommand(name = "scrape") {
 
     init {
-        this.subcommands(Catalog())
+        this.subcommands(Catalog(), Section(), Sections())
     }
 
     override fun run() = Unit
 
     /**
+     * CLI for performing search queries for catalog's section of NYU Albert based registration number.
+     */
+    private class Section() : CliktCommand(name = "section") {
+        private val logger = KotlinLogging.logger("scrape.section")
+        private val term: Term by option("--term").convert {
+            Term.fromId(Integer.parseInt(it))
+        }.required()
+        private val registrationNumber: Int by option(
+                "--registrationNumber",
+                help = "The registration number that you'd use to register for the course."
+        ).int().required()
+        private val file: String? by option("--file")
+        private val batchSize: Int? by option("--batchSize").int()
+        private val prettyPrint by option("--pretty").flag(default = false)
+
+        override fun run() {
+            val startTime = System.nanoTime()
+            file.writeToFileOrStdout(
+                    JsonMapper.toJson(
+                            scrapeFromSection(logger, term, registrationNumber), prettyPrint)
+            )
+
+            val endTime = System.nanoTime()
+            val duration = (endTime - startTime) / 1000000000.0 //divide by 1000000 to get milliseconds.
+            logger.info { "$duration seconds" }
+
+        }
+    }
+
+    /**
+     * CLI for performing search queries for catalog's sections of NYU Albert based on school and subject.
+     */
+    private class Sections() : CliktCommand(name = "sections") {
+        private val logger = KotlinLogging.logger("scrape.catalog")
+        private val term: Term by option("--term").convert {
+            Term.fromId(Integer.parseInt(it))
+        }.required()
+        private val school: String? by option("--school")
+        private val subject: String? by option("--subject")
+        private val file: String? by option("--file")
+        private val batchSize: Int? by option("--batchSize").int()
```

Comment:
I'm not sure batch size is being used right now; that's probably totally fine, because we'll probably want to include it in the future, but just a heads up that it's not being used rn.

Reply:
I keep it there to keep it consistent with the code you wrote. But since I don't really understand how it works, for now, I haven't used it for anything. I'm keeping there for the future, will prob make a note

Reply:
Ok, that's fine. The batchSize flag is for perf and not for anything else, so no worries we can keep it and fix later when we migrate the CLI to java.